File: c++/leetcode/645. Set Mismatch.cpp
class Solution
{
public:
  vector<int> findErrorNums(vector<int> &nums)
  {
    int dupe = -1, missing = 1;

    for (auto i : nums)
    {
      if (nums[abs(i) - 1] < 0)
        dupe = abs(i);
      else
        nums[abs(i) - 1] *= -1;
    }

    for (int i = 1; i < nums.size(); i++)
    {
      if (nums[i] > 0)
        missing = ++i;
    }

    return vector<int>{dupe, missing};
  }
};

File: javascript/leetcode/645. Set Mismatch.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var findErrorNums = function (nums) {
  let dupe = -1;
  let missing = 1;

  for (let i of nums) {
    if (nums[Math.abs(i) - 1] < 0) dupe = Math.abs(i);
    else nums[Math.abs(i) - 1] *= -1;
  }

  for (let i = 1; i < nums.length; i++) {
    if (nums[i] > 0) missing = ++i;
  }

  return [dupe, missing];
};
